src/App.tsx

```diff
@@ -1,116 +1,81 @@
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<Error | null | unknown>(null);
+  const [itemsPerPage] = React.useState(20);
+  const [searchQuery, setSearchQuery] = React.useState<string>('');
+  const [isShowMoreButtonDisable, setIsShowMoreButtonDisable] = React.useState<boolean>(false);
+  const [currentPage, setCurrentPage] = React.useState(0);
+  const [nextPage, setNextPage] = React.useState<number | null>(null);
+  const [prevPage, setPrevPage] = React.useState<number | null>(null);
+  const [isCardItemsDarked, setIsCardItemsDarked] = React.useState<boolean>(false);
+
+  const navigate = useNavigate();
+
+  const loadShows = async (page: number) => {
     if (isLoading) {
       return;
     }
 
-    this.setState({ isLoading: true });
+    setIsLoading(true);
 
     try {
-      const response = await fetch(`${URL.SHOWS}?page=${currentPage}`);
+      const response = await fetch(`${URL.SHOWS}?page=${page}`);
 
       if (!response.ok) {
         throw new Error('Network response error');
       }
 
       const data: ShowData[] = await response.json();
 
-      this.setState({
-        shows: data,
-        currentPage: currentPage === 8 ? currentPage - 8 : currentPage + 1,
-        isLoading: false,
-      });
+      setShows(data);
+      setIsLoading(false);
+      setNextPage(page + 1);
+      setPrevPage(page > 0 ? page - 1 : null);
+
+      navigate(`/${ROUTER_PATHS.SHOWS}?page=${encodeURIComponent(page)}`);
     } catch (error) {
-      this.setState({
-        shows: [],
-        isLoading: false,
-        error,
-      });
+      setShows([]);
+      setIsLoading(false);
+      setError(error);
     }
-  }
+  };
 
-  enableButton() {
-    const { shows } = this.state;
-
-    if (shows.length > 20) {
-      this.setState({ isShowMoreButtonDisable: true });
+  const enableButton = () => {
+    if (shows.length >= 20) {
+      setIsShowMoreButtonDisable(true);
     } else {
-      this.setState({ isShowMoreButtonDisable: false });
+      setIsShowMoreButtonDisable(false);
     }
-  }
-
-  showMoreShows() {
-    this.setState((prevState) => ({
-      isMoreShows: !prevState.isMoreShows,
-    }));
-  }
+  };
 
-  getSearchChange() {
+  const getSearchChange = () => {
     const savedValue = localStorage.getItem('TVShowSearch') || '';
-    this.setState({ searchQuery: savedValue });
-  }
+    setSearchQuery(savedValue);
+  };
 
-  async handleSearch() {
-    this.getSearchChange();
+  const handleSearch = async () => {
+    getSearchChange();
 
     const savedValue = localStorage.getItem('TVShowSearch') || '';
 
     if (savedValue === '') {
-      this.loadShows();
-      this.enableButton();
+      loadShows(currentPage);
+      enableButton();
       return;
     }
 
     try {
-      // const response = await fetch(`${URL.SEARCH}${searchQuery}`);
+      console.log(searchQuery);
       const response = await fetch(`${URL.SEARCH}${savedValue}`);
```

Comment:
It's a good practice to place api methods in a separate file and import it wherever you need:
e.g.
```
  export const getMovies = async (searchValue: string): Promise <...> => {
    try {
      const response = await fetch(`${URL.SEARCH}${searchValue}`);
      const data: ShowData[] = await response.json();
      
      return data;
    } catch (error) {
      throw error;
    }
  };
```
And here in the component you just call the API method in a try/catch block

Reply:
Thank you, I don't know why I didn't put it to a separate function in the first place, fixed in module04